### PR TITLE
Concourse pipeline

### LINKED ISSUE #1: Concourse pipeline to deploy Nessus

### PR DESCRIPTION
**Pull request by alice-carr**
This PR:
- adds a pipeline for concourse
- adds a function to generate api keys in the pipeline
- adds permissions for lambda role
- fixes installation of nessus on the instance

closes #1

### CODE REVIEW

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                terraform init
+                terraform plan
+                terraform apply --auto-approve
+                cd ..
+                pipenv run python generate_api_keys.py
+        # on_failure:
+        #   <<: *health_status_notify_for_prod
+        #   params:
+        #     message: "Failed to set SSH config"
+        #     health: unhealthy
```
Comment (akinnane) [inline concourse/pipeline.yml]:
```suggestion
         on_failure:
           <<: *health_status_notify_for_prod
           params:
             message: "Failed to set SSH config"
             health: unhealthy
```
I think we should re-enable the failure notifications.

File: generate_api_keys.py [hunk 2]
```diff
@@ -1,51 +1,176 @@
...
+        status = json.loads(response.text)
+        return status
+    except ConnectionError:
+        print("connection error")
+        return {"status": "loading", "progress": "0"}
+
+
+def main():
+    while True:
+        if get_status_checks():
```
Comment (akinnane) [inline generate_api_keys.py:156]: Can this ever fail? Do we need to have a timeout here?

File: generate_api_keys.py [hunk 3]
```diff
@@ -1,51 +1,176 @@
 import json
```
Comment (akinnane) [inline generate_api_keys.py:1]: No tests for this file? :(

File: concourse/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                cd cyber-security-nessus-git
+                pipenv --python 3.7
+                pipenv install
+
+                make zip
+                source /usr/local/bin/sts-assume-role.sh 'arn:aws:iam::676218256630:role/nessus_role' 'eu-west-2'
+                cd terraform
+                export TF_LOG=DEBUG
+                terraform init
+                terraform plan
```
[the remark this answers is not in the document]
Comment (alice-carr) [reply concourse/pipeline.yml:66]: Noted - removing this

File: generate_api_keys.py (hunk 3 above)
Comment (alice-carr) [reply generate_api_keys.py:1]: you're right - we'll do this right after this PR

File: concourse/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                pip install pipenv
+
+                cd cyber-security-nessus-git
+                pipenv --python 3.7
+                pipenv install
+
+                make zip
+                source /usr/local/bin/sts-assume-role.sh 'arn:aws:iam::676218256630:role/nessus_role' 'eu-west-2'
+                cd terraform
+                export TF_LOG=DEBUG
```
Comment (detnon) [inline concourse/pipeline.yml]: You may want to delete this, it'll be pretty noisy in concourse otherwise

Comment (detnon) [inline concourse/pipeline.yml]: You may want to delete `export TF_LOG=DEBUG`, otherwise it might be abit noisey

File: generate_api_keys.py (hunk 3 above)
Comment (pritchyspritch) [reply generate_api_keys.py:1]: This will be done next as part of #10

File: generate_api_keys.py (hunk 2 above)
Comment (alice-carr) [reply generate_api_keys.py:157]: I've added a timeout here

File: concourse/pipeline.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (alice-carr) [reply concourse/pipeline.yml]: thanks, I've removed this

Comment (alice-carr) [reply concourse/pipeline.yml]: thanks Sam - just removed this

File: concourse/pipeline.yml (hunk 1 above)
Comment (alice-carr) [reply concourse/pipeline.yml]: cool - I've done this

**Pull request merged**